src/utils/rest-api.ts

```diff
@@ -0,0 +1,250 @@
+                return false;
+            } else {
+                throw error;
+            }
+        });
+}
+
+function fetchEnv(): Promise<EnvJson> {
+    return fetch('env.json').then((res: ReqResponse) => res.json());
+}
+
+export function fetchAuthorizationCodeFlowFeatureFlag(): Promise<boolean> {
+    console.info(`Fetching authorization code flow feature flag...`);
+    return fetchEnv()
+        .then((env: EnvJson) =>
+            fetch(`${env.appsMetadataServerUrl}/authentication.json`)
+        )
+        .then((res: ReqResponse) => res.json())
+        .then((res: Record<string, any>) => {
+            console.log(
+                `Authorization code flow is ${
+                    res.authorizationCodeFlowFeatureFlag
+                        ? 'enabled'
+                        : 'disabled'
+                }`
+            );
+            return res.authorizationCodeFlowFeatureFlag;
+        })
+        .catch((error) => {
+            console.error(error);
+            console.warn(
+                `Something wrong happened when retrieving authentication.json: authorization code flow will be disabled`
+            );
+            return false;
+        });
+}
+
+export function fetchVersion(): Promise<Record<string, any>> {
+    console.info(`Fetching global metadata...`);
+    return fetchEnv()
+        .then((env: EnvJson) =>
+            fetch(`${env.appsMetadataServerUrl}/version.json`)
+        )
+        .then((response: ReqResponse) => response.json())
+        .catch((reason) => {
+            console.error(`Error while fetching the version : ${reason}`);
+            return reason;
+        });
+}
+
+export function fetchAppsAndUrls(): Promise<Array<Record<string, any>>> {
+    console.info(`Fetching apps and urls...`);
+    return fetchEnv()
+        .then((env: EnvJson) =>
+            fetch(`${env.appsMetadataServerUrl}/apps-metadata.json`)
+        )
+        .then((response: ReqResponse) => response.json());
+}
+
+export type ConfigParameter = {
+    //TODO check with config-server swagger
+    name: string;
+    value: any;
+    [propertiesName: string]: unknown; //temporary
+};
+export type ConfigParameters = Array<ConfigParameter>;
+export function fetchConfigParameters(
+    appName: string
+): Promise<ConfigParameters> {
+    console.info(`Fetching UI configuration params for app : ${appName}`);
+    const fetchParams = `${PREFIX_CONFIG_QUERIES}/v1/applications/${appName}/parameters`;
+    return backendFetchJson(fetchParams);
+}
+
+export function fetchConfigParameter(
+    name: string
+): ReturnType<typeof backendFetchJson> {
+    const appName = getAppName(name);
+    console.info(`Fetching UI config parameter '${name}' for app '${appName}'`);
+    const fetchParams = `${PREFIX_CONFIG_QUERIES}/v1/applications/${appName}/parameters/${name}`;
+    return backendFetchJson(fetchParams);
+}
+
+export function updateConfigParameter(
+    name: string,
+    value: Parameters<typeof encodeURIComponent>[0]
```

Comment:
It's just a string for us

Reply:
not necessarily
```ts
function encodeURIComponent(uriComponent: string | number | boolean): string
```

Reply:
I would simplify this type based on our needs and not on what encodeURIComponet method suggests, if one day we try to pass a type that is not handled by the method we will have a ts error in the method

Reply:
If you want to make a lib with wrappers of other lib methods I understand you don't want to reduce the use of the method and make it as generic as possible but here we're just building our application so our needs are the final needs already